### ISSUE TITLE
LoginResponse: Challenge required.

### ISSUE THREAD

**Issue by maswahid28** [JRoy/InstagramLive-PHP#9]
**Error While Logging in to Instagram: InstagramAPI\Response\LoginResponse:** _Challenge required_. "
I don't know why this error happen. But, probably the devs need to add Waiting Function during instagram check "This Was Me" on instagram app. Or some pending function when goLive.php requesting stream key to API server.

**Comment by JRoy** [comment]
This is because Instagram detects a new location, please sign in from your local network or try signing in from the web on the vps you are on. VPNs/Proxies are not supported

**Comment by maswahid28** [comment]
I dont not use proxies, I use same network with my phone and PC. But the error is still exist. Until I make new account in same network, I works. I think the script only works to new account. because When I use that for my old account it does'nt work . Maybe we can try script Two factor authentication .

https://github.com/mgp25/Instagram-API/blob/master/examples/twoFactorLogin.php

`<?php
set_time_limit(0);
date_default_timezone_set('UTC');
require __DIR__.'/../vendor/autoload.php';
/////// CONFIG ///////
$username = '';
$password = '';
$debug = true;
$truncatedDebug = false;
//////////////////////
$ig = new \InstagramAPI\Instagram($debug, $truncatedDebug);
try {
    $loginResponse = $ig->login($username, $password);
    if ($loginResponse !== null && $loginResponse->isTwoFactorRequired()) {
        $twoFactorIdentifier = $loginResponse->getTwoFactorInfo()->getTwoFactorIdentifier();
        // The "STDIN" lets you paste the code via terminal for testing.
        // You should replace this line with the logic you want.
        // The verification code will be sent by Instagram via SMS.
        $verificationCode = trim(fgets(STDIN));
        $ig->finishTwoFactorLogin($username, $password, $twoFactorIdentifier, $verificationCode);
    }
} catch (\Exception $e) {
    echo 'Something went wrong: '.$e->getMessage()."\n";`

**Comment by JRoy** [comment]
Try logging into instagram.com on the pc you're trying to on this on

**Comment by JRoy** [comment]
As for 2FA, I can implement that

**Comment by maswahid28** [comment]
Ok, I have try to login from my web on pc and login is success. But I had not try to live yet. Thank you so much for your spirit to fix and add this function. _May god bless you._

**Comment by rifqilia** [comment]
yeah, me and my team also got same problem, hope you can fix this immediately 😊

**Comment by JRoy** [comment]
Instructions for Challenge has been added with: 096317f

2FA Support has been added with: 88a8864

If you do enable 2FA I think that may disable the challenge but I have not tested that.

**Comment by JRoy** [comment]
Redownload the script and tell me how things go @maswahid28 @rifqilia

**Comment by maswahid28** [comment]
I still got account flagged . Give me more time to try again.

**Comment by maswahid28** [comment]
Account Flagged: Please sign out of all phones and try logging into instagram.com from this computer before trying to run this script again! 
------
Sign Out All of device : **Done**
Logging Into my PC on the same network : **Done**
No Proxy, No VPS, No VPN : **Done**
Change Password to make sure log out : **Done**

I don't know why this error still exist, maybe Instagram banns me because suspicious login activity before ?

**Comment by JRoy** [comment]
@maswahid28 try to enable 2 factor authentication on the account that may fix it

**Comment by maswahid28** [comment]
Again : Account Flagged
--------
I have enable two factor authentication . And the problem still exist in old account. But works perfectly in New Account. Well.... We have done our best. 😊😊😊

![image](https://user-images.githubusercontent.com/20895285/42922395-9305363a-8b4a-11e8-9809-3ff921481ab3.png)
_Account Flagged: Please sign out of all phones and try logging into instagram.com from this computer before trying to run this script again!_

![image](https://user-images.githubusercontent.com/20895285/42922450-c34aa19a-8b4a-11e8-862a-a43184c694dc.png)
_Suspicious Login Attempt_ 

![image](https://user-images.githubusercontent.com/20895285/42922478-dc200b42-8b4a-11e8-9b1f-72b8e903609c.png)
_Two Factor Authentication_

**Comment by JRoy** [comment]
Well this is out if my control, Instagram does this not me so :/

**Comment by maswahid28** [comment]
And..... Finally , it works.....
----------------------
I have been successfully live with my old account . I don't know how it can be. But, here what I have done so the tutorial works in my old account . 

**First :**  _I try to use USB Tethering so my PC connected to internet  from my phone. And my Phone was be like a USB WiFi adapter for my PC, Because I use WiFi network in my home .  Also I activate 2FA , so I log out all of devices, and only login to Instagram Website from Firefox installed on my PC._ 

**Second :** _The step above I think was my trick to avoid **(AFL)** Account Flagged, or **(CQ)** Challenge Required response from server, in other case several hours ago, my friend told me , he still got **(CQ)** although he was activate 2FA . And when I order him to do like what I did , the issues was still exist._ 

**Third** : _I don't know what I named it. But I mention it as "Trusted Network or Trusted Device" . When I feel very happy because Instagram have trust my network or device and to see this tutorial works, I try to disconnect my phone, and directly connect my PC to WiFi router . And When I type `php golive.php`  I didn't need any 2fa again and server give me stream key directly without worry about (AFL) or (CQ) from server._  

I think that's All from me.... So, what do you think, any improvement should be implemented...?

**Comment by vinstyle** [comment]
It's happening with me please find a solution i'd really appreciate that. I tried almost everything switching to two factor authentication and login and login out from other device but same message coming

**Comment by ZoranNik91** [comment]
I have this problem after running command : php -g goLive.php 

Loading InstagramLive-PHP v1.5.1...
Update: Outdated Instagram-API version detected, attempting to fix this for you. This may take a while...
'C:\Users\examp\Desktop\instagram' is not recognized as an internal or external command,
operable program or batch file.
Update: Finished! Exiting the script, please re-run the script now.

**Comment by brandnewguys** [comment]
> I have this problem after running command : php -g goLive.php
> 
> Loading InstagramLive-PHP v1.5.1...
> Update: Outdated Instagram-API version detected, attempting to fix this for you. This may take a while...
> 'C:\Users\examp\Desktop\instagram' is not recognized as an internal or external command,
> operable program or batch file.
> Update: Finished! Exiting the script, please re-run the script now.

We're having the same issue on MacOS. Seems like the API headers are not up to date. Seems like an easy fix?
